### PR TITLE
Rebase wiki

### LINKED ISSUE #1293: Redirected pages not listed in wildcard tag listing
This has been marked as a good candidate for becoming a [first-timers-only issue](http://www.firsttimersonly.com) like [these](https://github.com/publiclab/plots2/labels/first-timers-only), meaning that it's simple, self-contained, and with some extra formatting, could be a great entry point for a new contributor. If you're familiar enough with this code, please consider [reformatting or reposting it as a first-timers-only issue](https://publiclab.org/notes/warren/10-31-2016/create-a-welcoming-first-timers-only-issue-to-invite-new-software-contributors), and then ping @jywarren to get it labelled. Or, if this is not your first time, try to solve it yourself!

****

### Background

Some pages have a tag like `redirect:pagename` where `pagename` is the name of another page which the viewer should be automatically redirected to. Documentation:

https://publiclab.org/wiki/power-tags (look for `redirect`)

This behavior is captured (at the controller level) in this test:

https://github.com/publiclab/plots2/blob/master/test/functional/wiki_controller_test.rb#L507-L516

### Problem

These redirected pages should be listed on this tag listing page for wiki pages, using the wildcard `redirect:*` -- but nothing is shown there:

https://publiclab.org/wiki/tag/redirect:*

There are definitely redirected pages, like this one:

https://publiclab.org/wiki/camera-triggering

### Next steps

We should write a functional test to ensure that such tags appear on that tag listing page. There is already one for notes, but the separate action to display wiki pages is not tested:

https://github.com/publiclab/plots2/blob/master/test/functional/tag_controller_test.rb#L160-L167

I believe it uses the same action, but changes the `params[:node_type]` parameter:

https://github.com/publiclab/plots2/blob/master/app/controllers/tag_controller.rb#L23-L25

Once the test correctly describes this bug, changes to the tag controller will be able to resolve the problem.

### PR DESCRIPTION
**Pull request by StlMaris123**
fixes #1293, replaces #1357

* [x] all tests pass -- `rake test:all`
* [x] code is in uniquely-named feature branch, and has been rebased on top of latest master (especially if you've been asked to make additional changes)
* [x] pull request is descriptively named with #number reference back to original issue

Please be sure you've reviewed our contribution guidelines at https://publiclab.org/wiki/contributing-to-public-lab-software

We have a loose schedule of reviewing and pulling in changes every Tuesday and Friday, and publishing changes on Fridays. Please alert developers on plots-dev@googlegroups.com when your request is ready or if you need assistance.

Thanks!

### CODE REVIEW

File: Gemfile [hunk 1]
```diff
@@ -61,6 +61,7 @@ gem 'will_paginate', '>= 3.0.6'
 gem 'will_paginate-bootstrap', '>= 1.0.1'
 gem 'georuby', '2.0'
 gem 'geokit-rails'
+gem 'json'
```
Comment (jywarren) [inline Gemfile]: Oops, is this required? Also, could you remove the Gemfile.lock from the PR as well? Thanks!

File: test/functional/tag_controller_test.rb [hunk 2]
```diff
@@ -161,7 +161,24 @@ def setup
...
   end
 
-  test 'should show a featured wiki page at top, if it exists' do
+  test "wildcard tag show wiki pages" do
+    get :show, id: 'activities:*', node_type: 'wiki'
+    assert :success
+    assert_not_nil :tags
+    assert :wildcard
+    assert :wikis
+
```
Comment (jywarren) [inline test/functional/tag_controller_test.rb]: Let's remove the extra space here, sorry to be so detailed!

File: app/controllers/tag_controller.rb [hunk 3]
```diff
@@ -40,12 +44,16 @@ def show
                   .page(params[:page])
                   .order('node_revisions.timestamp DESC')
     end
+
+      # breaks the parameter after its rub
```
Comment (jywarren) [inline app/controllers/tag_controller.rb]: Can you align this comment and also, I'm not sure what you mean by "rub"? Thanks for the good documentation though!

**Comment by jywarren** [comment]
This is looking good -- i left a few requests, including especially the Gemfile and Gemfile.lock request (let's not include those if not essential!) and this should be good to go! Thanks for your patience on this!!!

**Comment by jywarren** [comment]
Super, just need to remove the Gemfile and Gemfile.lock changes and this will be ready to merge. Thanks, @StlMaris123 -- and don't hesitate to alert me with `@jywarren` if I miss something!

**Comment by jywarren** [comment]
Hi StellaMaris! Just checking in here, were you able to remove the extra commits?  https://stackoverflow.com/questions/215718/reset-or-revert-a-specific-file-to-a-specific-revision-using-git

**Comment by StlMaris123** [comment]
@jywarren Yes I removed

**Comment by StlMaris123** [comment]
@jywarren Yes I did. Please review and tell me what you think

File: Gemfile (hunk 1 above)
Comment (jywarren) [inline Gemfile]: Hi, @StlMaris123 - sorry, i still see this Gemfile change. Can you try one more time to remove this change and we can merge it?

File: Gemfile.lock [hunk 4]
```diff
@@ -95,9 +95,9 @@ GEM
     grape-entity (0.5.1)
       activesupport
       multi_json (>= 1.3.2)
-    grape-swagger (0.23.0)
+    grape-swagger (0.25.3)
```
Comment (jywarren) [inline Gemfile.lock]: Yes, see there are still changes to `Gemfile.lock` showing here. Did you `git push` your changes up once you removed it?

File: .ruby_version [hunk 5]
```diff
@@ -0,0 +1 @@
+2.1.2
```
Comment (jywarren) [inline .ruby_version]: Hmm, can you also remove this file?

File: Gemfile.lock (hunk 4 above)
Comment (jywarren) [inline Gemfile.lock]: Ah, can you remove all the changes to this file as well? Thank you, StellaMaris!

**Comment by StlMaris123** [comment]
All the files have been removed now

**Comment by jywarren** [comment]
Perfect, looks great!!! Merging this now :-)

**Pull request merged**